### ISSUE TITLE
systemd-nspawn: MACVLAN network interface, interface name not valid

### ISSUE THREAD

**Issue by tni** [systemd/systemd#28844]
### systemd version the issue has been seen with

254.1-1

### Used distribution

Arch

### Linux kernel version used

6.4.10

### CPU architectures issue was seen on

x86_64

### Component

systemd-nspawn

### Unexpected behaviour you saw

I'm getting an error message from systemd-nspawn:
MACVLAN network interface, interface name not valid: mv-enp0s20f0u2c2

The interface does exist (USB ethernet adapter), "ip address":
3: enp0s20f0u2c2: <BROADCAST,MULTICAST,UP,LOWER_UP> mtu 1500 qdisc fq_codel state UP group default qlen 1000

systemd-nspawn parameter used:
--network-macvlan=enp0s20f0u2c2

There is another network interface "eno2", which does work. "enp0s20f0u2c2" was working with systemd 253.

**Comment by mrc0mmand** [comment]
This is a regression caused by 2f091b1b495, as now we're parsing `enp0s20f0u2c2` as an interface pair `enp0s20f0u2c2:mv-enp0s20f0u2c2`, where the MACVLAN interface has 16 characters and trips over the `ifname_valid(b)` check in `network_iface_pair_parse()`. The old (v253) version didn't pass the ifnames through the `network_iface_pair_parse()` function and they got eventually properly shortened, e.g.:

```
# systemd-nspawn -D cont --network-macvlan=enp0s20f0u2c2
Spawning container cont on /home/mrc0mmand/repos/@systemd/systemd/cont.
Press Ctrl-] three times within 1s to kill container.
Using default interface naming scheme 'v253'.
Network interface name 'mv-enp0s20f0u2c2' has been changed to 'mv-enp0s20fc4OB' to fit length constraints.
...
```

**Comment by yuwata** [comment]
Fix is waiting in #28931.
